src/Microsoft.Azure.AppConfiguration.AspNetCore/AzureAppConfigurationRefreshMiddleware.cs

```diff
@@ -24,9 +25,15 @@ public AzureAppConfigurationRefreshMiddleware(RequestDelegate next, IConfigurati
 
         public async Task InvokeAsync(HttpContext context)
         {
-            foreach (var refresher in Refreshers)
+            //
+            // Configuration refresh is meant to execute as an isolated background task.
+            // To prevent access of request-based resources, such as HttpContext, we suppress the execution context within the refresh operation.
```

Comment:
`using (var flowControl = ExecutionContext.SuppressFlow())`

Can you use an explicit type. It's hard to tell what type `flowControl` is from the right hand side.